### PR TITLE
SER11-J. Prevent overwriting of externalizable objects

### PR DESCRIPTION
**Pull request by Spatrik95**
I extended the `SerializableIdiom` detector to check `readExternal()` function for the SER11-J bug.  
I created several test classes to make sure that every assigned class variable are need to be in an if condition. Also I checked that the variable in the if condition will get a new value. (so the initialized variable in the example changes value)  
I ran the extended detector for some larger repositories like ElasticSearch, but I didn't get any false-positive finding.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs/etc/messages.xml [hunk 1]
```diff
@@ -4544,6 +4544,16 @@ it may indicate a misunderstanding of how serialization works.
   <p> This class contains a field that is updated at multiple places in the class, thus it seems to be part of the state of the class. However, since the field is marked as transient and not set in readObject or readResolve, it will contain the default value in any
 deserialized instance of the class.
 </p>
+]]>
+    </Details>
+  </BugPattern>
+  <BugPattern type="SE_PREVENT_EXT_OBJ_OVERWRITE">
+    <ShortDescription>Prevent overwriting of externalizable objects</ShortDescription>
+    <LongDescription>Any caller can reset the value of the object by using the readExternal() method.</LongDescription>
+    <Details>
```
Comment (ThrawnCA) [inline spotbugs/etc/messages.xml:4553]: This doesn't say anything about how the issue might be resolved.

Comment (Spatrik95) [reply spotbugs/etc/messages.xml:4553]: I add a solution suggestion section to the description.

Comment (hazendaz) [reply spotbugs/etc/messages.xml:4559]: > This doesn't say anything about how the issue might be resolved.

Are you good with the proposed solution?  I think it reads ok generally but what are your thoughts?

@Spatrik95 I fixed the change log merge conflict for you.  If @ThrawnCA is good I Think we can merge in.

**Review by hazendaz — APPROVED** [review]

File: spotbugs/etc/messages.xml (hunk 1 above)
Comment (hazendaz) [reply spotbugs/etc/messages.xml:4567]: I went ahead and merged.  Thanks.

**Pull request merged**